### PR TITLE
[Site] Added Mailing List

### LINKED ISSUE #205: [Site] Mailing Lists: Add mailing list sign up to site
**Current State of the service-mesh-performance site**
SMP's mailing list and newsletter is not publicly available.

**Desired State of the service-mesh-performance site**
Similar to the Meshery site, the smp-spec.io site should have a newsletter sign-up and  the following mailing lists:

 - community@smp-spec.io - a form-based sign-up; open for anyone to signup for news / update on the project and service mesh topics in general.
 - user-group@smp-spec.io - a link to the google group; open for anyone to join.
 - maintainers@smp-spec.io - a link to the email address; open for anyone to send an email to.

### PR DESCRIPTION
**Pull request by NikhilSharma03**
Signed-off-by: Nikhil Sharma <nikhilsharmamusic2000@gmail.com>

**Description**
- Added Mailing List to the SMP site
- Maintained the responsiveness for small screens

Screenshot from 2021-07-23 17-48-19

Screenshot from 2021-07-23 17-48-48

This PR fixes #205

**Notes for Reviewers**
- Please check if the links are correct
- Let me know if there are any other changes we can make

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by warunicorn19** [comment]
Hey @NikhilSharma03 , So this looks great, good job. But, I feel like this is a little too big, might just be because of the picture.
@leecalcote what do you think ?

**Comment by NikhilSharma03** [comment]
@warunicorn19  Big as in for smaller screens?

**Comment by warunicorn19** [comment]
Overall, like maybe make it the same size as the newsletter signup?

**Comment by NikhilSharma03** [comment]
## After Changes
Screenshot from 2021-07-24 18-17-29

Screenshot from 2021-07-24 18-19-29

@warunicorn19 Need help here. Let me know if there are other changes we can make.

File: docs/_includes/subscribe.html [hunk 1]
```diff
@@ -35,7 +35,19 @@ <h3 class="section-subheading text-muted">{{ site.data.sitetext.subscribe.text |
...
 
         </div>
+
       </div>
+
+      <!--Mailing List-->
+      <div class="mailing-container">
+        <h1>Mailing Lists</h1>
+        <p>Engage in the Service Mesh Performance project. Join any of our mailinglists.</p>
+        <div class="mailing-btn-container">
```
Comment (warunicorn19) [inline docs/_includes/subscribe.html]: could you please put a gap between mailing and lists because they are two separate words.

**Review by warunicorn19 — CHANGES_REQUESTED** [review]

**Comment by warunicorn19** [comment]
Other changes look good to me, thanks @NikhilSharma03. Let's see what @leecalcote says.

File: docs/_includes/subscribe.html (hunk 1 above)
Comment (NikhilSharma03) [reply docs/_includes/subscribe.html]: Sorry, Missed this one. I will fix it immediately.

Comment (NikhilSharma03) [reply docs/_includes/subscribe.html]: Fixed.

File: docs/_includes/subscribe.html [hunk 2]
```diff
@@ -35,6 +35,19 @@ <h3 class="section-subheading text-muted">{{ site.data.sitetext.subscribe.text |
...
+
+      </div>
+
+      
+
+    </div>
+    <!--Mailing List-->
+    <div class="mailing-container">
+      <h1>Mailing Lists</h1>
+      <p>Engage in the Service Mesh Performance project. Join any of our mailing lists.</p>
```
Comment (leecalcote) [inline docs/_includes/subscribe.html:47]:
```suggestion
      <p>Engage in the Service Mesh Performance project. Join any of the mailing lists.</p>
```

File: docs/_includes/subscribe.html [hunk 3]
```diff
@@ -35,6 +35,19 @@ <h3 class="section-subheading text-muted">{{ site.data.sitetext.subscribe.text |
...
+      <div class="mailing-btn-container">
+          <a href="https://groups.google.com/a/smp-spec.io/d/forum/community" class="btn btn-primary btn-xl text-uppercase ">User</a>
```
Comment (leecalcote) [inline docs/_includes/subscribe.html:49]:
```suggestion
          <a href="https://groups.google.com/a/smp-spec.io/d/forum/community" class="btn btn-primary btn-xl text-uppercase ">Users</a>
```

File: docs/_includes/subscribe.html [hunk 4]
```diff
@@ -35,6 +35,19 @@ <h3 class="section-subheading text-muted">{{ site.data.sitetext.subscribe.text |
...
+          <a href="mailto:maintainers@smp-spec.io" class="btn btn-primary btn-xl text-uppercase ">Maintainer</a>
```
Comment (leecalcote) [inline docs/_includes/subscribe.html:50]:
```suggestion
          <a href="mailto:maintainers@smp-spec.io" class="btn btn-primary btn-xl text-uppercase ">Maintainers</a>
```

**Review by leecalcote — CHANGES_REQUESTED** [review]
Thank you, @nikhilsharma03. Will you update the hyperlinks for these buttons to be that of the hyperlinks provided in the issue description?

If we do away with the black background and try to make the mailing lists feel more like they are peers of the newsletter sign up form, I wonder if that might be an improvement.

**Comment by NikhilSharma03** [comment]
> Thank you, @NikhilSharma03. Will you update the hyperlinks for these buttons to be that of the hyperlinks provided in the issue description?
> 
> If we do away with the black background and try to make the mailing lists feel more like they are peers of the newsletter sign up form, I wonder if that might be an improvement.

Yeah Sure, I will make the necessary changes and then we can decide which one to go with. is that okay?

File: docs/_includes/subscribe.html (hunk 4 above)
Comment (NikhilSharma03) [reply docs/_includes/subscribe.html:50]: I will fix these.

**Comment by NikhilSharma03** [comment]
## Description
- After removing the black background and giving it the same color as the newsletter form.

## Screenshot
Screenshot from 2021-07-25 13-17-21

@leecalcote What do you think?

File: docs/_includes/subscribe.html (hunk 4 above)
Comment (leecalcote) [reply docs/_includes/subscribe.html:50]: Thanks. You can just click to commit these suggestions or batch commit the suggestions.

**Comment by leecalcote** [comment]
@JubayerJoy @Neilblaze one of you might want to review...

File: docs/_includes/subscribe.html (hunk 4 above)
Comment (NikhilSharma03) [reply docs/_includes/subscribe.html:50]: yeah sure, I will commit these changes.  Are we going with Dark Black bg color or the same as newsletter?

File: docs/_includes/subscribe.html [hunk 5]
```diff
@@ -35,6 +35,19 @@ <h3 class="section-subheading text-muted">{{ site.data.sitetext.subscribe.text |
...
+
+      </div>
+
+      
+
+    </div>
+    <!--Mailing List-->
+    <div class="mailing-container">
+      <h1>Mailing Lists</h1>
+      <p>Engage in the Service Mesh Performance project. Join any of the mailing lists.</p>
```
Comment (JubayerJoy) [inline docs/_includes/subscribe.html:47]: To keep consistency with the newsletter sub-heading
```suggestion
      <p class="text-muted">Engage in the Service Mesh Performance project. Join any of the mailing lists.</p>
```

**Review by JubayerJoy — COMMENTED** [review]
The implementation looks okay but it can be much more improved if the code follows the same structure as the newsletter section, such as using flex would be a good idea.
 
As per aesthetics, I think 3 primary buttons in a signal section feel a bit off. I'd like to refer to my all-time favorite [https://meshery.io/](https://meshery.io/) subscribe section. It sneaks three buttons, yet none seems out of place. I think @NikhilSharma03 can get much-needed inspiration from this. 

Screenshot (280)


What do you think @leecalcote?
Sidenote, shouldn't there be a space in "mailing list" at mesery?
If so, new `good-first-issue` 🥳

File: docs/_includes/subscribe.html (hunk 5 above)
Comment (Neilblaze) [reply docs/_includes/subscribe.html:47]: Yep, I think this sounds better! But Idk why it kind of seems like we're wasting so much space for those two sections. For responsive view, it's good to go as Jubayer said, & for the default preview, @NikhilSharma03 u can experiment with different aesthetic approaches to make it more eye-catching.

Comment (NikhilSharma03) [reply docs/_includes/subscribe.html:47]: Thank you for reviewing. I'll make the necessary changes and will get back to you.

**Comment by NikhilSharma03** [comment]
** Changes Made **
- Add `p` class
- Increased NL button width
- Removed Extra spacing 

** After Changes **
Screenshot from 2021-08-01 16-27-51

@JubayerJoy Can you please review?

**Comment by JubayerJoy** [comment]
> ** Changes Made **
> 
> * Added `p` class
> * Increased NL button width
> * Removed Extra spacing
> 
> ** After Changes **
> Screenshot from 2021-08-01 16-27-51
> 
> @JubayerJoy Can you please review?

@NikhilSharma03
Definitely an improvement on aesthetic 🎈
Please commit the change for review.
Thank you.

File: docs/_includes/subscribe.html (hunk 5 above)
Comment (JubayerJoy) [reply docs/_includes/subscribe.html:47]: I couldn't agree more with @Neilblaze.
Please see what more can be done for default.
Good luck 👍

**Comment by NikhilSharma03** [comment]
@JubayerJoy I have commited the recent changes. Can you please review?

File: docs/_sass/layout/_subscribe.scss [hunk 6]
```diff
@@ -3,7 +3,7 @@ section#subscribe {
   background-color: $gray-900;
   background-image: url("#{$subscribe-image}");
   padding-top: 50px;
-  padding-bottom: 80px;
+  // padding-bottom: 80px;
```
Comment (JubayerJoy) [inline docs/_sass/layout/_subscribe.scss:6]: It can be removed.
```suggestion
```

**Review by JubayerJoy — COMMENTED** [review]
LGTM 👍
Good work.

File: docs/_sass/layout/_subscribe.scss (hunk 6 above)
Comment (NikhilSharma03) [reply docs/_sass/layout/_subscribe.scss:6]: Missed It. I will fix it right away.

Comment (NikhilSharma03) [reply docs/_sass/layout/_subscribe.scss]: - [x] Fixed

**Comment by leecalcote** [comment]
Thank you, gents. Are we ✅ approved for merge here?

**Review by Jashpatel1 — COMMENTED** [review]
@NikhilSharma03 The links are not correct. Please fix them

**Comment by NikhilSharma03** [comment]
> @NikhilSharma03 The links are not correct. Please fix them

These links were given in the issue desc :sweat_smile: . Can you please drop the correct links. Thank you.

File: docs/_includes/subscribe.html [hunk 7]
```diff
@@ -34,8 +34,23 @@ <h3 class="section-subheading text-muted">{{ site.data.sitetext.subscribe.text |
  
           </form>
 
+          <!--Mailing List-->
+          <div class="mailing-container">
+            <h1>Mailing Lists</h1>
+            <p  class="text-muted">Engage in the Service Mesh Performance project. Join any of the mailing lists.</p>
+            <div class="mailing-btn-container">
+                <a href="user-group@smp-spec.io" class="btn btn-primary btn-xl text-uppercase ">Users</a>
```
Comment (sayantan1413) [inline docs/_includes/subscribe.html:42]:
```suggestion
                <a href="https://groups.google.com/a/smp-spec.io/g/user-group" class="btn btn-primary btn-xl text-uppercase ">Users</a>
```

File: docs/_includes/subscribe.html [hunk 8]
```diff
@@ -34,8 +34,23 @@ <h3 class="section-subheading text-muted">{{ site.data.sitetext.subscribe.text |
...
+                <a href="maintainers@smp-spec.io" class="btn btn-primary btn-xl text-uppercase ">Maintainers</a>
```
Comment (sayantan1413) [inline docs/_includes/subscribe.html:43]:
```suggestion
                <a href="https://groups.google.com/a/smp-spec.io/g/maintainers" class="btn btn-primary btn-xl text-uppercase ">Maintainers</a>
```

**Review by sayantan1413 — CHANGES_REQUESTED** [review]
Hey @NikhilSharma03 I think we are done with the PR just change the links and we are good to go!

File: docs/_includes/subscribe.html (hunk 8 above)
Comment (NikhilSharma03) [reply docs/_includes/subscribe.html]: @sayantan1413 Can you please check?

**Comment by sayantan1413** [comment]
Hey @NikhilSharma03 I saw that in this [line](https://github.com/service-mesh-performance/service-mesh-performance/blob/c249688659839050e4077e1cf9de4cb45995e2dc/docs/_includes/subscribe.html#L28) after the onclick = a '(' is missing can you check that once

**Comment by sayantan1413** [comment]
@Jashpatel1 can you verify if the links are okay now and if anyother changes are required here ?  // @warunicorn19

**Comment by NikhilSharma03** [comment]
> Hey @NikhilSharma03 I saw that in this [line](https://github.com/service-mesh-performance/service-mesh-performance/blob/c249688659839050e4077e1cf9de4cb45995e2dc/docs/_includes/subscribe.html#L28) after the onclick = a '(' is missing can you check that once

Sharp Eyes :laughing: . I think it was done in any other PR since I haven't made any changes in the form. I will fix it right away.

**Comment by warunicorn19** [comment]
That's what was causing the test to fail😅.

**Review by warunicorn19 — APPROVED** [review]
if @Jashpatel1 has nothing further to add, these changes look good to me. Thanks, @NikhilSharma03 for seeing this to the end.

**Review by sayantan1413 — APPROVED** [review]

File: docs/_includes/subscribe.html [hunk 9]
```diff
@@ -34,8 +34,23 @@ <h3 class="section-subheading text-muted">{{ site.data.sitetext.subscribe.text |
  
           </form>
 
+          <!--Mailing List-->
+          <div class="mailing-container">
+            <h1>Mailing Lists</h1>
+            <p  class="text-muted">Engage in the Service Mesh Performance project. Join any of the mailing lists.</p>
+            <div class="mailing-btn-container">
+                <a href="https://groups.google.com/a/smp-spec.io/g/user-group" class="btn btn-primary btn-xl text-uppercase ">Users</a>
+                <a href="https://groups.google.com/a/smp-spec.io/g/maintainers" class="btn btn-primary btn-xl text-uppercase ">Maintainers</a>
```
Comment (leecalcote) [inline docs/_includes/subscribe.html:43]:
```suggestion
                <a href="https://groups.google.com/a/smp-spec.io/g/community" class="btn btn-primary btn-xl text-uppercase ">Community</a>
```

**Review by leecalcote — APPROVED** [review]

**Pull request merged**